src/mempolicy.c

```diff
@@ -45,6 +48,22 @@ umf_result_t umfMempolicySetInterleavePartSize(umf_mempolicy_handle_t policy,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    policy->ops.part_size = partSize;
+    policy->ops.interleave.part_size = partSize;
+    return UMF_RESULT_SUCCESS;
+}
+
+umf_result_t
+umfMempolicySetCustomSplitPartitions(umf_mempolicy_handle_t policy,
+                                     umf_mempolicy_split_partition_t *partList,
+                                     size_t partListLen) {
+    if (policy == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
```

Comment:
we could have tests for these 2 negative scenarios

Reply:
Done